cluster-scope/bundles/gpu-taint-management/taint.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# GPU Taint Watcher Script
+
+# Initial tainting of GPU nodes
+kubectl get nodes -l node-role.kubernetes.io/worker -o json |
+jq -r '
+  .items[]
+  | select(.metadata.labels["nvidia.com/gpu.product"] != null)
+  | .metadata.name + " " + .metadata.labels["nvidia.com/gpu.product"]
+' | while read NODE PRODUCT; do
+  echo "Initial tainting: $NODE with GPU product $PRODUCT"
+  kubectl taint node "$NODE" "nvidia.com/gpu.product=$PRODUCT:NoSchedule" --overwrite=false 2>/dev/null
+done
+
+# Declare associative array to track last seen GPU product per node
+declare -A LAST_PRODUCT
+
+# Continuous monitoring loop
+while true; do
+  kubectl get nodes -l node-role.kubernetes.io/worker -o json |
+  jq -r '.items[].metadata.name' | while read NODE_NAME; do
+    PRODUCT=$(kubectl get node "$NODE_NAME" -o jsonpath="{.metadata.labels['nvidia\.com/gpu\.product']}" 2>/dev/null)
+
+    # Skip nodes without GPU product label
+    if [ -z "$PRODUCT" ]; then
+      echo "No GPU product label on $NODE_NAME, skipping"
+      continue
+    fi
```

Comment:
Instead of listing all workers and then ignoring those without the `nvidia.com/gpu.product` label, why not just add `-l nvidia.com/gpu.product` to the first `kubectl` invocation?

```
kubectl get nodes -l node-role.kubernetes.io/worker -l nvidia.com/gpu.product -o name | cut -f2 -d/ |
while read NODE_NAME; do
    # Check current taints on the node
    CURRENT_TAINT=$(kubectl get node "$NODE_NAME" -o jsonpath="{.spec.taints[?(@.key=='nvidia.com/gpu.product')].value}" 2>/dev/null)
    ...
done
```